returnn/tf/layers/basic.py

```diff
@@ -3771,7 +3774,7 @@ def get_out_data_from_opts(cls, name, axis, dim=1, sources=(), **kwargs):
       data = data.copy_as_batch_major()
     axis = cls._get_axis(data=data, axis=axis)
 
-    new_dim = SpatialDim("%s_expand_dims" % name, dim)
+    new_dim = (FeatureDim if init_axis.lower() == "f" else SpatialDim)("%s_expand_dims" % name, dim)
```

Comment:
I personally would have preferred to use `Dim(kind=... if ... else ..., description=..., dimension=dim)`. But maybe that's just personal preference.

Reply:
I agree it's a bit nicer what you have, changed it.